test/test_index_management.py

```diff
@@ -50,12 +56,170 @@ def test_inputs(self):
+
+        # Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until ``listSearchIndexes`` returns an empty array.
+        t0 = time.time()
+        while True:
+            indices = list(coll0.list_search_indexes())
+            if indices:
+                break
+            if (time.time() - t0) / 60 > 5:
+                raise TimeoutError("Timed out waiting for index deletion")
+            time.sleep(5)
+
+    def test_case_4(self):
+        """Driver can update a search index."""
+        # Create a collection with the "create" command using a randomly generated name (referred to as ``coll0``).
+        coll0 = self.db[f"col{uuid.uuid4()}"]
+        coll0.insert_one({})
+
+        # Create a new search index on ``coll0``.
+        model = {"name": _NAME, "definition": {"mappings": {"dynamic": False}}}
+        resp = coll0.create_search_index(model)
+
+        # Assert that the command returns the name of the index: ``"test-search-index"``.
+        self.assertEqual(resp, _NAME)
+
+        # Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+        #  An index with the ``name`` of ``test-search-index`` is present and index has a field ``queryable`` with the value of ``true``.
+        self.wait_for_ready(coll0)
+
+        # Run a ``updateSearchIndex`` on ``coll0``.
+        # Assert that the command does not error and the server responds with a success.
+        model2: dict[str, Any] = {"name": _NAME, "definition": {"mappings": {"dynamic": True}}}
+        coll0.update_search_index(_NAME, model2["definition"])
+
+        # Run ``coll0.listSearchIndexes()`` repeatedly every 5 seconds until the following condition is satisfied:
+        #   An index with the ``name`` of ``test-search-index`` is present.  This index is referred to as ``index``.
+        # The index has a field ``queryable`` with a value of ``true`` and has a field ``status`` with the value of ``READY``.
+        predicate = lambda index: index.get("queryable") is True and index.get("status") == "READY"
+        self.wait_for_ready(coll0, predicate=predicate)
+
+        # Assert that an index is present with the name ``test-search-index`` and the definition has a property ``latestDefinition`` whose value is ``{ 'mappings': { 'dynamic': true } }``.
+        index = list(coll0.list_search_indexes(_NAME))[0]
+        self.assertIn("latestDefinition", index)
+        self.assertEqual(index["latestDefinition"], model2["definition"])
+
+    def test_case_5(self):
+        """``dropSearchIndex`` suppresses namespace not found errors."""
+        # Create a driver-side collection object for a randomly generated collection name.  Do not create this collection on the server.
+        coll0 = self.db[f"col{uuid.uuid4()}"]
+
+        # Run a ``dropSearchIndex`` command and assert that no error is thrown.
+        coll0.drop_search_index("foo")
+
+
+if os.environ.get("TEST_INDEX_MANAGEMENT"):
```

Comment:
Shouldn't these tests also run against a regular cluster so that we test the `"isError": true,` cases?

Reply:
Yes, it is running against the second cluster we started with mongo-orchestration.  We use the env variable to make sure we don't accidentally run when we're using an Atlas cluster as the `MONGODB_URI` on another test run.

Reply:
In that case I think it would be better to remove the if-statement here.

Reply:
No, because the test wouldn't error as expected if the file is run against a 7.0 Atlas cluster.  We will need https://jira.mongodb.org/browse/DRIVERS-2688 to avoid the workaround.

Reply:
Interesting, I was not aware we run these tests against Atlas at all. What task fails when this is run? It would be great to allow these tests to run against a local cluster without needing to set an env var.

Reply:
Also I would settled for a comment explaining the issue and referencing DRIVERS-2688 but still curious to know which EVG tasks fail.

Reply:
Good point, I verified that none of our existing Atlas tests would fail, but added the guard and a note.

Reply:
So reading your comment I still think we can remove this guard because we don't run these tests against atlas.

Reply:
Done